packages/next/test/fixtures/00-node-middleware/middleware.js

```diff
@@ -0,0 +1,256 @@
+import { NextResponse } from 'next/server';
+
+const ALLOWED = ['allowed'];
+
+export const config = {
+  runtime: 'nodejs',
+  matcher: [
+    '/dynamic/:path*',
+    '/_sites/:path*',
+    '/:teamId/:slug',
+    '/:path*',
+    '/',
+  ],
+};
+
+export function middleware(request) {
+  const url = request.nextUrl;
+  const pathname = url.pathname;
+
+  if (process.env.FOO) {
+    console.log(`Includes env variable ${process.env.FOO}`);
+  }
+
+  if (url.pathname === '/redirect-me') {
+    url.pathname = '/from-middleware';
+    return NextResponse.redirect(url, 307);
+  }
+
+  if (url.pathname === '/next') {
+    return NextResponse.next();
+  }
+
+  if (url.pathname === '/version') {
+    return NextResponse.json({
+      enumerable: Object.keys(self).includes('VercelRuntime'),
+      version: self.VercelRuntime.version,
+    });
+  }
+
+  if (url.pathname === '/globals') {
+    const globalThisKeys = Object.keys(globalThis);
+    const globalKeys = globalThisKeys.reduce((acc, globalName) => {
+      const key = globalName.toString();
+      if (global[key]) acc.push(key);
+      return acc;
+    }, []);
+
+    const res = NextResponse.next();
+    res.headers.set(
+      'data',
+      JSON.stringify({ globals: globalKeys, globalThis: globalThisKeys })
+    );
+    return res;
+  }
+
+  if (url.pathname === '/log') {
+    console.log('hi there');
+    return;
+  }
+
+  if (url.pathname === '/somewhere') {
+    url.pathname = '/from-middleware';
+    return NextResponse.redirect(url);
+  }
+
+  if (url.pathname === '/logs') {
+    console.clear();
+    for (let i = 0; i < 3; i++) console.count();
+    console.count('test');
+    console.count('test');
+    console.dir({ hello: 'world' });
+    console.log('hello');
+    console.log('world');
```

Comment:
this too I think?  or if we're asserting on the logging behavior, i didn't see a test